tests/test_deserialization.py

```diff
@@ -1178,6 +1178,18 @@ class AliasingUserSerializer(Schema):
         assert result["email"] == "foo@bar.com"
         assert "years" not in result
 
+    def test_deserialize_dotted_path_with_data_key_param(self):
+        class MySchema(Schema):
+            foo = fields.Str(data_key="foo")
+            bar = fields.Str(data_key="data.bar")
+
+        sch = MySchema(unknown=EXCLUDE)
+        data = {"foo": "dotted", "data": {"bar": "paths!"}}
+
+        ret = sch.load(data)
+        assert ret["foo"] == "dotted"
+        assert ret["bar"] == "paths!"
+
```

Comment:
Nitpick: `dotted` is a bit misleading. At first I thought it was meant to suggest that the `foo` field is accessed with dotted notation, while it is the other one. It took me a few reads to realize it is meant to be read as "dotted" + "paths!" = "dotted paths!".